android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java

```diff
@@ -0,0 +1,32 @@
+package com.liferay.mobile.screens.base;
+
+import android.util.Log;
+
+import com.liferay.mobile.screens.context.LiferayServerContext;
+
+/**
+ * @author Javier Gamarra
+ */
+public class FactoryProvider {
+
+	public static AbstractFactory getInstance() {
+		if (_abstractFactory == null) {
+			synchronized ("factory") {
```

Comment:
Are you sure this synchronization is right? If "factory" is a new String object for every call, then it won't never block